tests/core/vm/test_computation.py

```diff
@@ -0,0 +1,92 @@
+from eth_typing import Address
+
+from eth_utils import decode_hex
+
+from eth.chains.base import MiningChain
+from eth.chains.mainnet import MAINNET_VMS
+
+from eth import constants
+from eth.consensus import NoProofConsensus
+from eth.exceptions import (
+    InvalidInstruction,
+)
+
+
+def _configure_mining_chain(starting_vm, vm_under_test):
+    return MiningChain.configure(
+        __name__="AllVMs",
+        vm_configuration=(
+            (
+                constants.GENESIS_BLOCK_NUMBER,
+                starting_vm.configure(consensus_class=NoProofConsensus),
+            ),
+            (
+                constants.GENESIS_BLOCK_NUMBER + 1,
+                vm_under_test.configure(consensus_class=NoProofConsensus),
+            ),
+        ),
+        chain_id=1337,
+    )
+
+
+# CREATE, RETURNDATASIZE, and RETURNDATACOPY opcodes not added until Byzantium
+@pytest.fixture(params=MAINNET_VMS[4:])
+def byzantium_plus_miner(request, base_db, genesis_state):
+    byzantium_vm = MAINNET_VMS[4]
+    vm_under_test = request.param
+
+    klass = _configure_mining_chain(byzantium_vm, vm_under_test)
+    header_fields = dict(
+        difficulty=1,
+        gas_limit=100000,  # arbitrary, just enough for testing
+    )
+    return klass.from_genesis(base_db, header_fields, genesis_state)
+
+
+@pytest.mark.parametrize(
+    "code",
+    (
+        # generate some return data, then call CREATE (third-to-last opcode - 0xf0)
+        decode_hex('0x5b595958333d5859858585858585858585f195858585858585858485858585f195858585858585f1f03d30'),  # noqa: E501
+        # generate some return data, then call CREATE2 (third-to-last opcode - 0xf5)
+        decode_hex('0x5b595958333d5859858585858585858585f195858585858585858485858585f195858585858585f1f53d30'),  # noqa: E501
+    )
+)
+def test_CREATE_and_CREATE2_resets_return_data_if_account_has_insufficient_funds(
+    byzantium_plus_miner,
+    canonical_address_a,
+    transaction_context,
+    code,
+):
+    chain = byzantium_plus_miner
+    vm = chain.get_vm()
+    state = vm.state
+
+    assert state.get_balance(canonical_address_a) == 0
+
+    computation = vm.execute_bytecode(
+        origin=canonical_address_a,
+        to=canonical_address_a,
+        sender=Address(
+            b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x12\x34'
+        ),
+        value=0,
+        code=code,
+        data=code,
+        gas=400000,
+        gas_price=1,
+    )
+
+    if computation.is_error:
+        assert isinstance(computation.error, InvalidInstruction)
+        # only test CREATE case for byzantium as the CREATE2 opcode (0xf5) was not yet introduced
+        assert vm.fork == "byzantium"
+        assert "0xf5" in repr(computation.error).lower()
+
+    else:
+        # Assert only uses the appropriate amount of gas.
+        # This varies between the CREATE (34925) and CREATE2 (36725) tests.
+        assert computation.get_gas_used() == 34925 or computation.get_gas_used() == 36725
```

Comment:
Maybe you could pass in the expected gas usage as a parametrized value?

I assume this test is to make sure that we don't burn the remaining gas. That could be added as a comment (since it may not be obvious why the gas used is related to the return data value)

Reply:
For sure. I think that would be better.

Reply:
It actually turns out the discrepancy was due to the gas estimation changes in Berlin. We don't particularly care to test the exact gas value here since that's not what the scope of this test is so I am just going to assert that we do not burn all the gas and that it falls within an expected range here.